datafusion/sql/src/planner.rs

```diff
@@ -61,6 +61,15 @@ pub trait ContextProvider {
         not_impl_err!("Table Functions are not supported")
     }
 
+    /// TODO: add doc
+    fn create_cte_work_table(
```

Comment:
I think the `ContextProvider` is designed so the parser doesn't depend on the rest of the DataFusion execution machinery so it can supply table providers

However, in this case it seems to me like the work tables aren't going to come from a context provider (they will be made once per query)

Thus I wonder if this function would be better simply as a function on the `SqlToRel ` planner itself

Reply:
yeah I think that makes sense. I wasn't too keen on expanding the ContextPlanner interface to workaround the crate graph either.

Reply:
I'm not sure of a way to directly implement this method on the `SqlToRel` because the `sql` code needs access to a table that contains `execution` related types that can't be a direct dependency of the `sql` crate (namely, the CteWorktable (see https://github.com/matthewgapp/arrow-datafusion/blob/68c9fe7ddc0f1c5bbc1d011c6d4d34c8d73ca656/datafusion/core/src/datasource/cte_worktable.rs#L38). The context provider was a workaround to hide the execution types behind the context provider.

Reply:
Maybe we can just add a comment about the rationale and that would be good enough

Reply:
will do

Reply:
done